### PR TITLE
add feature flag on sign out with save and resume

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2150

### Change description

add feature flag on sign out with save and resume and hide help details when feature flag off

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/sign-out.html [hunk 1]
```diff
@@ -66,6 +66,7 @@
         classes: "govuk-visually-hidden"
       }) }}
 
+      {% if saveAndResume %}
       <details class="govuk-details" data-module="govuk-details" data-event-id="find-saved-information-details">
```
Comment (mwestacott) [inline views/sign-out.html:70]: Please could you indent lines 70 to 87 by two spaces, just to improve readability.

File: test/controllers/sign.out.controller.spec.ts [hunk 2]
```diff
@@ -60,8 +60,35 @@ describe("Sign Out controller", () => {
...
+      mockIsActiveFeature.mockReturnValueOnce(true);
+      const resp = await request(app)
+        .get(`${config.SIGN_OUT_URL}?page=${config.SOLD_LAND_FILTER_PAGE}`);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(SIGN_OUT_PAGE_TITLE);
+      expect(resp.text).toContain(SIGN_OUT_HINT_TEXT);
       expect(resp.text).toContain(SIGN_OUT_HELP_DETAILS_TEXT);
+      expect(resp.text).toContain(SIGN_OUT_DROPDOWN_TEXT);
+      expect(resp.text).toContain('How do I find my saved applications?');
```
Comment (mwestacott) [inline test/controllers/sign.out.controller.spec.ts:77]: Please define this string with an appropriate constant name (`SIGN_OUT_SAVE_AND_RESUME_GUIDANCE_TEXT`?) in the 'text.mock.ts' file and reference it from there (both here and below). I think you also need to check for the presence of the following string in this test and that it's not present in your other new test below (again, define as a constant, `SIGN_OUT_SAVE_AND_RESUME_GUIDANCE_DETAILS_TEXT` perhaps):

_"Sign in to your Companies House account and select Your filings to access your saved applications."_

**Review by mwestacott — COMMENTED** [review]
@AndreKramer7654 you may wish to review this as it also impacts - fixes a bug on - the update journey when save and resume is not active.

File: test/controllers/sign.out.controller.spec.ts (hunk 2 above)
Comment (HatiDev) [reply test/controllers/sign.out.controller.spec.ts:79]: added

File: test/controllers/sign.out.controller.spec.ts [hunk 3]
```diff
@@ -71,8 +73,8 @@ describe("Sign Out controller", () => {
 
       expect(resp.status).toEqual(200);
       expect(resp.text).toContain(SIGN_OUT_PAGE_TITLE);
-      expect(resp.text).toContain(SIGN_OUT_HINT_TEXT);
-      expect(resp.text).toContain(SIGN_OUT_HELP_DETAILS_TEXT);
+      expect(resp.text).toContain(SIGN_OUT_SAVE_AND_RESUME_GUIDANCE_TEXT);
```
Comment (mwestacott) [inline test/controllers/sign.out.controller.spec.ts:76]: I wasn't expecting these two lines to change, rather that line 79 was modified to reference the new constant and then a new 'expect' line added below with the second constant.

File: test/controllers/sign.out.controller.spec.ts [hunk 4]
```diff
@@ -85,8 +87,8 @@ describe("Sign Out controller", () => {
 
       expect(resp.status).toEqual(200);
       expect(resp.text).toContain(SIGN_OUT_PAGE_TITLE);
-      expect(resp.text).not.toContain(SIGN_OUT_HINT_TEXT);
-      expect(resp.text).not.toContain(SIGN_OUT_HELP_DETAILS_TEXT);
+      expect(resp.text).not.toContain(SIGN_OUT_SAVE_AND_RESUME_GUIDANCE_TEXT);
```
Comment (mwestacott) [inline test/controllers/sign.out.controller.spec.ts:90]: Same here. I would revert the changes in these two lines but change line 93 (and also add a new one) to reference your new constants. Thanks.

**Review by mwestacott — APPROVED** [review]
